### PR TITLE
fix: listview is not in the center of the widget

### PR DESCRIPTION
**Pull request by Decodetalkers**
set the listview size is the same as item count, and put it in the center.

And in commonbackgrounditem, not resize the image directly, keep the full shape of the origin image, final clip the center part as the shown image

Log:

Issue 没有找到。。我记得有的，而且我也不太爽这个地方

### CODE REVIEW

**Comment by Decodetalkers** [comment]
这个35意思是model的高度，似乎是自己推断的？

File: src/plugin-commoninfo/window/commonbackgrounditem.cpp [hunk 1]
```diff
@@ -142,13 +142,35 @@ void CommonBackgroundItem::updateBackground(const QPixmap &pixmap)
...
+    QSize finalSize = ((float)showSize.width() / (float)pixmapSize.width()) * pixmapSize;
+    qreal ratio = devicePixelRatioF();
+
+    auto previewPixmap = m_basePixmap.scaled(finalSize,
                                        Qt::IgnoreAspectRatio,
-                                       Qt::FastTransformation);
+                                       Qt::SmoothTransformation);
+
+
+    if (previewPixmap.size().height() <= showSize.height()) {
```
Comment (18202781743) [inline src/plugin-commoninfo/window/commonbackgrounditem.cpp:158]: 这段代码用if else 分支去处理previewPixmap吧

File: src/plugin-commoninfo/window/bootwidget.cpp [hunk 2]
```diff
@@ -34,6 +34,12 @@ Q_DECLARE_METATYPE(QMargins)
 const QMargins ListViewItemMargin(10, 8, 10, 8);
 const QVariant VListViewItemMargin = QVariant::fromValue(ListViewItemMargin);
 
+constexpr int LISTVIEW_SPACING = 1;
```
Comment (18202781743) [inline src/plugin-commoninfo/window/bootwidget.cpp]: 加上static修饰，

File: src/plugin-commoninfo/window/bootwidget.cpp [hunk 3]
```diff
@@ -253,11 +266,11 @@ void BootWidget::setEntryList(const QStringList &list)
...
 {
     int cout = m_bootList->count();
-    int height = (cout + 2) * 35;
+    int height = (cout + 2) * LISTVIEW_ITEM_HEIGHT;
 
-    m_listLayout->addWidget(m_bootList);
+    m_listLayout->addWidget(m_bootList, Qt::AlignCenter);
     m_listLayout->addSpacing(15);
-    m_background->setFixedHeight(height + 35 > 350 ? 350 : height + 35);
+    m_background->setFixedHeight(height + LISTVIEW_ITEM_HEIGHT > MAX_BACKGROUND_HEIGHT ? MAX_BACKGROUND_HEIGHT : height + LISTVIEW_ITEM_HEIGHT);
```
Comment (18202781743) [inline src/plugin-commoninfo/window/bootwidget.cpp:273]: 用max函数吧，

File: src/plugin-commoninfo/window/bootwidget.cpp [hunk 4]
```diff
@@ -34,6 +34,12 @@ Q_DECLARE_METATYPE(QMargins)
 const QMargins ListViewItemMargin(10, 8, 10, 8);
 const QVariant VListViewItemMargin = QVariant::fromValue(ListViewItemMargin);
 
+constexpr int LISTVIEW_ITEM_SPACING = 3;
+constexpr int LISTVIEW_ITEM_HEIGHT = 32;
+constexpr int LISTVIEW_ITEM_HEIGHT_WITH_SPACING = LISTVIEW_ITEM_SPACING + LISTVIEW_ITEM_HEIGHT;
```
Comment (18202781743) [inline src/plugin-commoninfo/window/bootwidget.cpp:39]: 用static修饰下吧，

**Review by 18202781743 — APPROVED** [review]

**Pull request merged**